src/mesh/tests/MeshTest.cpp

```diff
@@ -821,5 +821,288 @@ BOOST_AUTO_TEST_CASE(AddMesh)
   BOOST_TEST(globalMesh->tetrahedra().size() == 3);
 }
 
+BOOST_AUTO_TEST_CASE(RemoveDuplicates)
```

Comment:
I guess we test here the implicit primitive generation as well, right?

Reply:
True. I copy the edge deduplication into a separate test.